File: 2025/October-17-2025/twosumII.js
var nums = [1,2,3,4], target = 3

const twoSumII = (nums, target) => { 
  let left = 0; 
  let right = nums.length - 1; 

  while(left < right){
    let sum = nums[left] + nums[right]

    if(sum < target){
      left++
    } else if (sum > target){
      right--
    } else { 
      return [left + 1, right + 1]
    }
  }
  return []
}

console.log(twoSumII(nums, target))

File: 2025/November-7-2025/twoSumII.py
from typing import List

nums = [1,2,3,4]
target = 3

def two_sumii(nums: List[int], target: int) -> List[int]: 
  left = 0 
  res = []
  right = len(nums) - 1

  while left < right: 
    sum = nums[right] + nums[left]
    if sum < target: 
      left+=1
    elif sum > target: 
      right-=1
    else: 
      res = [nums[left], nums[right]]
      left+=1
      right-=1
  return res


print(two_sumii(nums, target))